code/espurna/relay.cpp

```diff
@@ -1457,8 +1473,30 @@ void _relaySetupAdhoc() {
     #endif
 
     _relays.reserve(relays);
-    for (unsigned char id = 0; id < relays; ++id) {
-        _relays.emplace_back(id);
+
+    #if (RELAY_PROVIDER == RELAY_PROVIDER_RELAY)
+        using gpio_type = GpioPin;
+    #elif (RELAY_PROVIDER == RELAY_PROVIDER_MCP23S08)
+        using gpio_type = McpGpioPin;
+    #endif
```

Comment:
Regarding travis issue, I also needs to match RELAY_PROVIDER_LIGHT for every check I added